### PR TITLE
Docs for new field of Model Deployment "predictor"

### PR DESCRIPTION
**Pull request by keshamin**
Documented `predictor` field that allows different Inference Backends with different APIs and protocols.

### CODE REVIEW

File: source/ref_deployments.rst [hunk 1]
```diff
@@ -9,10 +9,34 @@ Additionally, it provides the following set of features:
...
+Predictors
+***************************
+
+A model can be deployed in ODAHU if only it is packed with a supported Inference Server (predictor).
+Inference Server is typically a web service that "wraps" an ML model and lets remote clients to invoke
+the model via HTTP (or any other protocol).
+
+ODAHU currently supports 2 predictors:
+
+- ODAHU ML Server: :code:`odahu-ml-server`
```
Comment (dsuslov) [inline source/ref_deployments.rst]: @keshamin odahu-ml-server -> odahu

File: source/ref_deployments.rst [hunk 2]
```diff
@@ -9,10 +9,34 @@ Additionally, it provides the following set of features:
...
+- NVIDIA Triton Server: :code:`triton`
+
+ODAHU ML Server
+################
+
+ODAHU ML Server is an inference server that build a simple HTTP layer on top of any
```
Comment (dsuslov) [inline source/ref_deployments.rst]: that build - that builds

File: source/ref_deployments.rst [hunk 3]
```diff
@@ -9,10 +9,34 @@ Additionally, it provides the following set of features:
...
+`MLFlow model <https://www.mlflow.org/docs/latest/models.html>`_ with an HTTP layer.
+
+To pack a model into ODAHU ML Server :ref:`Docker REST <ref_packagers:Docker REST>` packager can be used.
+
+NVIDIA Triton Server
+####################
+
+`Triton Server <https://github.com/triton-inference-server/server>`_ is a feature-rich inference server. To pack a model into a Triton Server,
+:ref:`Triton Packager <ref_packagers:Nvidia Triton Packager>` should be used.
```
Comment (dsuslov) [inline source/ref_deployments.rst]: should be used -> has to be used.

File: source/ref_deployments.rst [hunk 4]
```diff
@@ -9,10 +9,34 @@ Additionally, it provides the following set of features:
...
+- ODAHU ML Server: :code:`odahu-ml-server`
+- NVIDIA Triton Server: :code:`triton`
+
+ODAHU ML Server
+################
+
+ODAHU ML Server is an inference server that build a simple HTTP layer on top of any
+`MLFlow model <https://www.mlflow.org/docs/latest/models.html>`_ with an HTTP layer.
+
+To pack a model into ODAHU ML Server :ref:`Docker REST <ref_packagers:Docker REST>` packager can be used.
```
Comment (dsuslov) [inline source/ref_deployments.rst]: - ODAHU ML Server -> ODAHU inference server
- can be used -> has to be used

File: source/ref_deployments.rst (hunk 2 above)
Comment (dsuslov) [inline source/ref_deployments.rst]: ODAHU ML Server -> inference server

File: source/ref_deployments.rst [hunk 5]
```diff
@@ -9,10 +9,34 @@ Additionally, it provides the following set of features:
     * Scale to zero
     * Dynamic Model swagger
     * Monitoring of Model Deployment
-    
-.. important::
 
-   Model could be deployed only if docker image was packaged through Docker REST packager.
+***************************
+Predictors
```
Comment (dsuslov) [inline source/ref_deployments.rst]: Inference Servers

File: source/ref_deployments.rst [hunk 6]
```diff
@@ -9,10 +9,34 @@ Additionally, it provides the following set of features:
...
+***************************
+
+A model can be deployed in ODAHU if only it is packed with a supported Inference Server (predictor).
```
Comment (dsuslov) [inline source/ref_deployments.rst]: - A model -> A ML model
- Inference Server (predictor) -> Inference Server

File: source/ref_deployments.rst [hunk 7]
```diff
@@ -9,10 +9,34 @@ Additionally, it provides the following set of features:
...
+Inference Server is typically a web service that "wraps" an ML model and lets remote clients to invoke
+the model via HTTP (or any other protocol).
+
+ODAHU currently supports 2 predictors:
```
Comment (dsuslov) [inline source/ref_deployments.rst]: supports 2 predictors  ->supports several Inference Servers

File: source/ref_deployments.rst [hunk 8]
```diff
@@ -9,10 +9,34 @@ Additionally, it provides the following set of features:
...
+
+- ODAHU ML Server: :code:`odahu-ml-server`
+- NVIDIA Triton Server: :code:`triton`
+
+ODAHU ML Server
```
Comment (dsuslov) [inline source/ref_deployments.rst]: ODAHU ML Server -> ODAHU Inference Server

File: source/ref_deployments.rst [hunk 9]
```diff
@@ -9,10 +9,34 @@ Additionally, it provides the following set of features:
...
+################
+
+ODAHU ML Server is an inference server that build a simple HTTP layer on top of any
+`MLFlow model <https://www.mlflow.org/docs/latest/models.html>`_ with an HTTP layer.
+
+To pack a model into ODAHU ML Server :ref:`Docker REST <ref_packagers:Docker REST>` packager can be used.
+
+NVIDIA Triton Server
```
Comment (dsuslov) [inline source/ref_deployments.rst]: NVIDIA Triton Inference Server

File: source/tutorials_wine.rst [hunk 10]
```diff
@@ -616,13 +616,15 @@ Paste code into the file:
    kind: ModelDeployment
    spec:
      image: "<fill-in>"
+     predictor: odahu-ml-server
```
Comment (dsuslov) [inline source/tutorials_wine.rst:619]: odahu-mlflow

File: source/ref_deployments.rst [hunk 11]
```diff
@@ -9,10 +9,34 @@ Additionally, it provides the following set of features:
...
+####################
+
+`Triton Server <https://github.com/triton-inference-server/server>`_ is a feature-rich inference server. To pack a model into a Triton Server,
+:ref:`Triton Packager <ref_packagers:Nvidia Triton Packager>` should be used.
+Triton Server uses `KFServing Inference Protocol <https://github.com/kubeflow/kfserving/blob/master/docs/predict-api/v2/required_api.md>`_.
 
 ********************************************
 General deployment structure
```
Comment (dsuslov) [inline source/ref_deployments.rst]: General deployment structure -> General deployment manifest structure

File: source/ref_deployments.rst [hunk 12]
```diff
@@ -30,6 +54,10 @@ General deployment structure
     #  * end with an alphanumeric character
     id: wine-12345
     spec:
+        # Predictor is an inference backend name; required field
+        # Possible values are: odahu-ml-server, triton
+        predictor: odahu-ml-server
```
Comment (dsuslov) [inline source/ref_deployments.rst:66]: odahu-ml-server - odahu-mlflow

**Review by dsuslov — CHANGES_REQUESTED** [review]

File: source/tutorials_wine.rst (hunk 10 above)
Comment (keshamin) [reply source/tutorials_wine.rst:619]: @dsuslov 
Our Flask wrapper basically works with GPPI format, no matter if MLFlow inside or not. We can name it `odahu-gppi-server`.

**Review by dsuslov — APPROVED** [review]

**Review by vlad-tokarev — APPROVED** [review]

File: source/ref_deployments.rst [hunk 13]
```diff
@@ -9,13 +9,40 @@ Additionally, it provides the following set of features:
...
     * Monitoring of Model Deployment
-    
-.. important::
 
-   Model could be deployed only if docker image was packaged through Docker REST packager.
+***************************
+Inference Servers
+***************************
+
+A model can be deployed in ODAHU if only it is packed with a supported Inference Server.
```
Comment (karbyshevds) [inline source/ref_deployments.rst:17]:
```suggestion
A model can be deployed to ODAHU only if it is packed with a supported Inference Server.
```

**Review by karbyshevds — DISMISSED** [review]

**Pull request merged**